commons-util/src/main/kotlin/ebi/ac/uk/util/regex/PatternExtensions.kt

```diff
@@ -0,0 +1,16 @@
+package ebi.ac.uk.util.regex
+
+import arrow.core.Option
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+/**
+ * Try to match the pattern over the provided string and return optional of matcher.
```

Comment:
I'd suggest:  [The function] Tries to match the pattern over the provided string and returns an optional containing the matcher result.